include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -78,15 +81,68 @@ enum E_DESCRIPTOR_TYPE : uint32_t
 template<typename SamplerType>
 class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 {
+	struct SBindingRedirect
+	{
+		static constexpr inline uint32_t Invalid = ~0u;
+
+		SBindingRedirect() : bindings(nullptr), offsets(nullptr), count(0ull) {}
+
+		SBindingRedirect(const size_t _count) : count(_count)
+		{
+			bindings = std::make_unique<uint32_t[]>(count << 1);
+			offsets = bindings.get() + count;
+		}
+
+		// For a given binding number `binding` in `bindings`, return its offset in `offsets`.
+		inline uint32_t operator[](const uint32_t binding) const
+		{
+			const uint32_t index = searchForBinding(binding);
+			if (index == Invalid)
+				return Invalid;
+
+			return offsets[index];
+		}
+
+		// Returns index into the `bindings` and `offsets` array for the given binding number `binding.
+		// Assumes `bindings` is sorted.
+		inline uint32_t searchForBinding(const uint32_t binding) const
```

Comment:
make the method private

Reply:
actually all members, and only provide const getters, but make the layout a friend

Reply:
or instead of the friendship, move `SBindingRedirectBuildInfo` out of the layout constructor into the `private` section of the layout.

Then move the `buildRedirects` lambda to the second constructor of `SBindingRedirect`

Reply:
Resolved.